MBBSEmu/TextVariables/TextVariableService.cs

```diff
@@ -59,5 +64,105 @@ public int GetVariableIndex(string name)
 
             return -1;
         }
+
+        /// <summary>
+        ///     Parses incoming buffer to process text variables before sending to client
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="sessionValues"></param>
+        /// <returns></returns>
+        public ReadOnlySpan<byte> Parse(ReadOnlySpan<byte> input, Dictionary<string, string> sessionValues)
+        {
+            using var newOutputBuffer = new MemoryStream(input.Length);
+            for (var i = 0; i < input.Length; i++)
+            {
+                //Look for initial signature byte -- faster
+                if (input[i] != 0x1)
+                {
+                    newOutputBuffer.WriteByte(input[i]);
+                    continue;
+                }
+
+                //If we found a 0x1 -- but it'd take us past the end of the buffer, we're done
+                if (i + 3 >= input.Length)
+                    break;
+
+                //Look for justification notation in i + 1 = "N, L, C, R", if not found move on
+                if (input[i + 1] != 0x4E && input[i + 1] != 0x4C && input[i + 1] != 0x43 && input[i + 1] != 0x52)
+                    continue;
+
+                //Get formatting information
+                var variableFormatJustification = input[i + 1];
+                var variableFormatPadding = (input[i + 2] - 32);
+
+                //Increment 3 Bytes
+                i += 3;
+
+                var variableNameStart = i;
+                var variableNameLength = 0;
+
+                //Get variable name
+                while (input[i] != 0x1)
+                {
+                    i++;
+                    variableNameLength++;
+                }
+
+                var variableName = Encoding.ASCII.GetString(input.Slice(variableNameStart, variableNameLength));
+                var variableText = GetVariableByName($"{variableName}");
```

Comment:
I don't think we need the string formatter here, can just pass in `variableName`